plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/ASTStructureValidator.xtend

```diff
@@ -1225,9 +1225,11 @@ class ASTStructureValidator {
 		)
 	}
 
-	def private void validateLabel(Statement model, LabelledStatement label, ASTStructureDiagnosticProducer producer,
-		Set<LabelledStatement> validLabels) {
-		if (label !== null && !label.eIsProxy && !validLabels.contains(label)) {
+	def private void validateLabelRef(LabelRef model, ASTStructureDiagnosticProducer producer,
+		Set<LabelledStatement> validLabels
+	) {
+		val labelAsText = model.labelAsText; // cannot use model.label, because we aren't allowed to resolve proxies in this phase!
+		if (labelAsText !== null && !validLabels.exists[it.name==labelAsText]) {
```

Comment:
you can ignore that `it`, e.g.`name == labelAsText`

Reply:
added that on purpose to highlight that `labelAsText` is not a property of `it`

Reply:
ah OK :)